pkg/clusteragent/orchestrator/transformer.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build kubeapiserver,orchestrator
+
+package orchestrator
+
+import (
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/util/orchestrator"
+
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func extractDeployment(d *v1.Deployment) *model.Deployment {
+	deploy := model.Deployment{
+		Metadata: orchestrator.ExtractMetadata(&d.ObjectMeta),
+	}
+	// spec
+	deploy.ReplicasDesired = 1 // default
+	if d.Spec.Replicas != nil {
+		deploy.ReplicasDesired = *d.Spec.Replicas
+	}
+	deploy.Paused = d.Spec.Paused
+	deploy.DeploymentStrategy = string(d.Spec.Strategy.Type)
+	if deploy.DeploymentStrategy == "RollingUpdate" {
+		deploy.MaxUnavailable = d.Spec.Strategy.RollingUpdate.MaxUnavailable.StrVal
+		deploy.MaxSurge = d.Spec.Strategy.RollingUpdate.MaxSurge.StrVal
```

Comment:
Can't those spec fields be unspecified? I think we need to handle default values like it was done for `ReplicasDesired`.